### ISSUE TITLE
Implement SDL2 GameController Events

### ISSUE THREAD

**Issue by pathunstrom** [ppb/pursuedpybear#560]
Before we can do anything with joysticks and game controllers, we need the events handled by the translation layer.

Implement the maps from the SDL events to ppb events, defined as needed to support the GameController events.

Union documentation is here: http://wiki.libsdl.org/SDL_EventType?highlight=%28%5CbCategoryEnum%5Cb%29%7C%28CategoryEvents%29

We need to implement:

* SDL_CONTROLLERAXISMOTION
* SDL_CONTROLLERBUTTONDOWN
* SDL_CONTROLLERBUTTONUP
* SDL_CONTROLLERDEVICEADDED
* SDL_CONTROLLERDEVICEREMOVED
* SDL_CONTROLLERDEVICEREMAPPED

**Comment by AstraLuma** [comment]
Note that the interface should also reasonably map to https://developer.mozilla.org/en-US/docs/Web/API/Gamepad_API/Using_the_Gamepad_API

**Comment by AstraLuma** [comment]
Probably don't need both this and #559.

**Comment by AstraLuma** [comment]
Closing in favor #447
